BootloaderCommonPkg/Library/ShellLib/CmdBoot.c

```diff
@@ -403,13 +402,22 @@ PrintBootOption (
                  BootOption->Image[0].LbaImage.LbaAddr \
                  );
     }
-
-    for (ExtraIndex = 1; ExtraIndex < LoadImageTypeMax; ExtraIndex++) {
-      if (BootOption->Image[ExtraIndex].LbaImage.Valid == 1) {
-        ShellPrint (L" [%x|0x%x]", BootOption->Image[ExtraIndex].LbaImage.SwPart, \
-          BootOption->Image[ExtraIndex].LbaImage.LbaAddr);
-      } else if (BootOption->Image[ExtraIndex].FileName[0] != '\0') {
-        ShellPrint (L" [%a]", BootOption->Image[ExtraIndex].FileName);
+    //Print Pre-OS image filename

```

Comment:
can we use a loop to avoid code duplication for the print for PreOS and Extra image?

Reply:
we always print normal OS with ImageType, DevType, DevInstance, BootFlags, HwPart, FsType, SwPart and LbaAddr.
On top of it, we only print PreOS info if PreOS flag is set. and print Extra info if Extra flag is set.
So it might not be applicable to use a loop.

Reply:
I meant the code below line #406 to #420 are very close to each other.  It can be done using a loop of 2.

Reply:
Got it. Or we could create a function and call the function as below.
	PrintExtraImages (BootOption, BOOT_FLAGS_PREOS, LoadImageTypePreOs);
	PrintExtraImages (BootOption, BOOT_FLAGS_EXTRA, LoadImageTypeExtra0);

VOID
PrintExtraImages (
  OS_BOOT_OPTION             *BootOption,
  UINT8                      Flags,
  LOAD_IMAGE_TYPE            ImageType
  )
{
    if ((BootOption->BootFlags & Flags) != 0){
        if (BootOption->Image[ImageType].LbaImage.Valid == 1) {
          ShellPrint (L" [%x|0x%x]", ImageType, BootOption->Image[ImageType].LbaImage.SwPart, \
            BootOption->Image[ImageType].LbaImage.LbaAddr);
        } else if (BootOption->Image[ImageType].FileName[0] != '\0') {
          ShellPrint (L" [%a]", BootOption->Image[ImageType].FileName);
        }
    }
}

Reply:
yes, use function should be fine.

Reply:
updated patch